pkg/kubernetes/event_handlers.go

```diff
@@ -51,7 +51,15 @@ func GetKubernetesEventHandlers(informerName, providerName string, announce chan
 			ann.ReferencedObjectID = getObjID(obj)
 		}
 
-		announce <- ann
+		select {
+		case announce <- ann:
+			// Channel post succeeded
+		default:
+			// Since pubsub introduction, there's a chance we start seeing full channels which
+			// will slowly become unused in favour of pub-sub subscriptions.
+			// We are making sure here ResourceEventHandlerFuncs never locks due to a push on a full channel.
+			log.Debug().Msgf("Channel for provider %s is full, dropping channel notify %s ", providerName, eventType)
```

Comment:
I'd say - escalate that to `Error` -- if this ever happens - we'd want to know that it happened

Reply:
I'd not advice that for now, if you ask me. There's a number of channels that will quickly go full with a number of events that are not being read, purely because we will stop using them. I can remove them, but that's not what I intended at least in this patch for the sake of diff size.

We will very quickly populate logs with errors. I'd instead suggest to open an issue and track this to be cleaned up instead.

Reply:
Let's change this to `log.Warn()` so that we know this is unexpected but system should be ok.

Reply:
I'll address as requested, we'll discuss at a later time if we see this is too chatty.